src/GraphQL/Introspection/__Type.cs

```diff
@@ -43,13 +43,15 @@ public __Type()
                         var includeDeprecated = context.GetArgument<bool>("includeDeprecated");
                         var type = context.Source as IComplexGraphType;
                         var fields = !includeDeprecated
-                            ? type?.Fields.Where(f => string.IsNullOrWhiteSpace(f.DeprecationReason))
```

Comment:
`type` always non null here because both `IObjectGraphType` and `IInterfaceGraphType` inherit `IComplexGraphType`